packages/fields/src/types/File/Implementation.js

```diff
@@ -4,19 +4,17 @@ import { KnexFieldAdapter } from '@keystonejs/adapter-knex';
 import mongoose from 'mongoose';
 
 // Disabling the getter of mongoose >= 5.1.0
-// https://github.com/Automattic/mongoose/blob/master/migrating_to_5.md#checking-if-a-path-is-populated
+// https://mongoosejs.com/docs/migrating_to_5.html#id-getter
 mongoose.set('objectIdGetter', false);
 
 const {
   Types: { ObjectId },
 } = mongoose;
 
 export class File extends Implementation {
-  constructor(path, { directory, route, adapter }) {
+  constructor(path, { adapter }) {
     super(...arguments);
     this.graphQLOutputType = 'File';
-    this.directory = directory;
-    this.route = route;
```

Comment:
I think this should stay. It might be what's needed to fix #1684, and #1823. They are added to the Admin meta but not used in the field views. If we add the route to the image path in views it should allow users to configure a path for images. I'm not sure that directory is as useful.

Also as I write this I'm thinking now that maybe this info should be pulled off the `fileAdapter`.

Thoughts?

Reply:
Yeah, I'd say it it should probably be pulled off the adapter. It avoid having to have things configured in two places.